action.yml

```diff
@@ -128,3 +136,22 @@ runs:
       with:
         name: PyPI README
         path: /tmp/baipp/dist/out/sdist/PyPI-README.*
+
+    - name: >
+        Generate JSON objects to specifiy subsequent job matrixes with supported 
+        Python versions
+      id: supported-pythons-setter
+      shell: bash
+      working-directory: /tmp/baipp/dist/out/sdist/
+      run: |
+          cat */PKG-INFO | python -c '
+          import json, re, sys
+          match_classifier = re.compile(
+              r"\s*Classifier: Programming Language :: Python :: (\d+\.\d+)$"
```

Comment:
a correct implementation needs to consider requires_python and the list of supported python versions of githubs setup_python

Reply:
i could argue that this component shouldn't serve for deferred metadata consistency validation and generally as well as particularly regarding the second concerns: the targeted application environment is the one where we want things to crash in order to fix upon. the provided outputs are supposed to be building blocks, afaik `jq` is available in the runner environment.

Reply:
The reason the version list should come from requires-python is that that list Will be what pip will consider/respect

The trove classifiers are neither complete nor under mandatory construction

Reply:
Yeah, `requires-python` is the canonical metadata we ought to check. It would be a nice-to-have to raise an error, if it it contradicts the classifiers, but the classifiers are just a FYI.

That said there's the big question of how would we decide the upped Python version bound?

You get that from the classifiers which kinda made them an indicator of what the CI is checking against. But the supported upper bound would have to technically change whenever GitHub Actions add a new Python version?

Reply:
If no upper bound was given,then it's a good idea to get a failure as soon as a new python version gets added that fails

I'd consider known good upper bounds as out of scope for this one